torch/csrc/jit/tensorexpr/ir_verifier.cpp

```diff
@@ -119,7 +119,19 @@ void IRVerifier::visit(IfThenElsePtr v) {
 }
 
 void IRVerifier::visit(IntrinsicsPtr v) {
+  if (v->op_type() == kIsNan) {
+    if (v->dtype().scalar_type() != c10::kInt) {
+      throw malformed_ir("bad dtype in intrinsic arg");
+    }
+    IRVisitor::visit(v);
+    return;
+  }
   // TODO: add a check for OpArgCount and op_type
+  for (auto const& param : v->params()) {
+    if (param->dtype() != v->dtype()) {
+      throw malformed_ir("bad dtype in intrinsic arg");
+    }
+  }
```

Comment:
Is it a requirement that all parameters should have the same dtype as the result value? I guess it's true at the moment, but I wonder if it's appropriate to enforce this condition in IR verifier

Reply:
Yeah it's definitely not universally true (as you see I had to special-case `isnan`), but it's otherwise true for the set of intrinsics we have.  I think llvm would complain anywhere this doesn't hold, since the type signatures wouldn't match.